packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/DeleteRulesTests.java

```diff
@@ -0,0 +1,231 @@
+import org.alfresco.rest.exception.EmptyRestModelCollectionException;
+import org.alfresco.rest.model.RestRuleModel;
+import org.alfresco.rest.model.RestRuleModelsCollection;
+import org.alfresco.utility.model.FolderModel;
+import org.alfresco.utility.model.SiteModel;
+import org.alfresco.utility.model.TestGroup;
+import org.alfresco.utility.model.UserModel;
+import org.junit.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/**
+ * Tests for DELETE /nodes/{nodeId}/rule-sets/{ruleSetId}/rules/{ruleId}.
+ */
+@Test(groups = {TestGroup.RULES})
+public class DeleteRulesTests extends RestTest
+{
+    private static final String FAKE_NODE_REF = "fake-node-id";
+
+    private UserModel user;
+    private SiteModel site;
+
+    @BeforeClass(alwaysRun = true)
+    public void dataPreparation()
+    {
+        STEP("Create a user, site and folder");
+        user = dataUser.createRandomTestUser();
+        site = dataSite.usingUser(user).createPublicRandomSite();
+    }
+
+    /**
+     * Delete a rule by its id.
+     */
+    @Test(groups = {TestGroup.REST_API, TestGroup.RULES, TestGroup.SANITY})
+    public void deleteSingleRuleAndGet204()
+    {
+        STEP("Create rules in the folder");
+        final FolderModel ruleFolder = dataContent.usingUser(user).usingSite(site).createFolder();
+        final List<RestRuleModel> createdRules = Stream.of("ruleA", "ruleB", "ruleC")
+                .map(ruleName -> {
+                    RestRuleModel ruleModel = new RestRuleModel();
+                    ruleModel.setName(ruleName);
+                    return restClient.authenticateUser(user).withCoreAPI().usingNode(ruleFolder).usingDefaultRuleSet()
+                            .createSingleRule(ruleModel);
+                })
+                .collect(toList());
+
+        STEP("Attempt delete the rule ");
+        final RestRuleModel ruleA = createdRules.get(0);
+        restClient.authenticateUser(user).withCoreAPI().usingNode(ruleFolder).usingDefaultRuleSet().deleteRule(ruleA.getId());
+        restClient.assertStatusCodeIs(NO_CONTENT);
+
+        STEP("Get and check the rules from the folder after deleting one of them");
+        final RestRuleModelsCollection rulesAfterDeletion =
+                restClient.authenticateUser(user).withCoreAPI().usingNode(ruleFolder).usingDefaultRuleSet().getListOfRules();
+        restClient.assertStatusCodeIs(OK);
+        rulesAfterDeletion.assertThat().entriesListCountIs(createdRules.size() - 1);
+        Assert.assertTrue(rulesAfterDeletion.getEntries()
+                .stream()
+                .noneMatch(r -> r.onModel().getId().equals(ruleA.getId()))
+        );
+        rulesAfterDeletion.getEntries().stream()
+                .map(rulesLeft -> rulesLeft.onModel().getId())
+                .forEach(left -> Assert.assertTrue( createdRules.stream()
+                        .map(RestRuleModel::getId)
+                        .anyMatch(left::equals)
```

Comment:
This is fine, but it would be more accurate to just check the set of ids match.

Reply:
Good point - changed that.